File: Android App/app/src/main/java/revert/shredsheets/models/scales/Hungarian.java
package revert.shredsheets.models.scales;

public class Hungarian extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{ 3, 1, 2, 1, 2, 1, 2};
    }

    @Override
    public String getName() {
        return "Hungarian";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{"Hungarian", "SuperLocrian ♭♭6 ♭♭7", "Harmonic Minor ♭5", "SuperLocrian ♯6", "Jazz Minor ♯5", "Dorian ♭9 ♯11", "Lydian Augmented ♯3"};
    }

    @Override
    public int[][] getDegrees() {
        return new int[][] {
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 },
                { 1, 2, 3, 4, 5, 6, 7 }
        };
    }

    @Override
    public String[] getChords() {
        return new String[]{"7", "m7o", "mΔ7 ♭5", "m7ø", "mΔ7+", "m7", "Δ7+♯3"};
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/Hungarian.kt
package com.bm.shredsheets.models.scales

class Hungarian : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(3, 1, 2, 1, 2, 1, 2)

    override val name: String
        get() = "Hungarian"

    override fun getModeNames(): Array<String> {
        return arrayOf("Hungarian", "SuperLocrian ♭♭6 ♭♭7", "Harmonic Minor ♭5", "SuperLocrian ♯6", "Jazz Minor ♯5", "Dorian ♭9 ♯11", "Lydian Augmented ♯3")
    }

    override val getEnum: Scales
        get() = Scales.Hungarian

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7))

    override val chords: Array<String>
        get() = arrayOf("7", "m7o", "mΔ7 ♭5", "m7ø", "mΔ7+", "m7", "Δ7+♯3")
}